core/jazz_t-vault/components/utils/vault.js

```diff
@@ -0,0 +1,389 @@
+// =========================================================================
+
+const request = require("request");
+const logger = require('../logger.js');
+
+function getVaultToken(configData) {
+  return new Promise((resolve, reject) => {
+    let payload = {
+      uri: `${configData.T_VAULT_API}${global.globalConfig.API.LOGIN}`,
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "Accept": "application/json"
+      },
+      json: {
+        "username": configData.T_VAULT_USERNAME,
+        "password": configData.T_VAULT_PASSWORD
+      }
+    };
+
+    request(payload, function (error, response, body) {
+      if (response.statusCode && response.statusCode === 200 && body && body.client_token) {
+        logger.debug("Successfully logged into tvault");
+        return resolve(body.client_token);
+      } else {
+        logger.error("Error in getting vault token: " + JSON.stringify(response));
+        return reject({
+          "error": "Error in getting vault token: " + response.body.errors
+        });
+      }
+    });
+  });
+}
+
+function createSafe(safeDetails, configData, vaultToken, onComplete) {
+
+  let payload = {
+    uri: `${configData.T_VAULT_API}${global.globalConfig.API.SAFE}`,
+    method: "POST",
+    headers: {
+      "vault-token": vaultToken
+    },
+    json: {
+      "path": `${global.globalConfig.PATH_TO_SAFE}${safeDetails.name.toLowerCase()}`,
+      "data": {
+        "name": `${safeDetails.name.toLowerCase()}`,
+        "owner": `${safeDetails.owner}`,
+        "description": `${safeDetails.description}`
+      }
+    },
+    rejectUnauthorized: false
+  };
+
+
+  logger.debug("createSafe payload : " + JSON.stringify(payload));
+  request(payload, function (error, response, body) {
+    logger.debug("createSafe response : " + JSON.stringify(response));
+    if (response.statusCode && response.statusCode === 200 && body) {
+      logger.debug("Successfully created safe");
+      return onComplete(null, body);
+    } else {
+      logger.error("Error in creating safe: " + JSON.stringify(error));
+      return onComplete({
+        "error": "Error in creating safe: " + response.body.errors
+      });
+    }
+  });
+}
+
+function getSafeDetails(safename, configData, vaultToken, onComplete) {
+  let payload = {
+    uri: `${configData.T_VAULT_API}${global.globalConfig.API.SAFE}${global.globalConfig.API.GET_SAFE}${safename.toLowerCase()}`,
+    method: "GET",
+    headers: {
+      "vault-token": vaultToken
+    },
+    rejectUnauthorized: false
+  };
+
+  logger.debug("getSafeDetails payload : " + JSON.stringify(payload));
+  request(payload, function (error, response, body) {
+    logger.debug("getSafeDetails response: " + JSON.stringify(response));
+    if (response.statusCode && response.statusCode === 200) {
+      const safeDetails = JSON.parse(body);
+      logger.debug("Successfully got safe details");
+      return onComplete(null, safeDetails.data);
```

Comment:
I see that GET /safes/{safeName} returns safe details along with list of users & roles and corresponding permissions. However, I don't see any specific calls to vault getting those details or logic to massage the data coming from vault. 

Can you check if we are returning data as per the swagger spec?

Reply:
The T-vault response itself formatted, so there is no need for  further massage from our side. But applied the logic to convert "aws-role" key to roles key in response.